org.lflang/src/org/lflang/validation/LFValidatorImpl.xtend

```diff
@@ -549,7 +549,7 @@ class LFValidatorImpl extends AbstractLFValidator {
 
     @Check(FAST)
     def checkDeadline(Deadline deadline) {
-        if (this.target == Target.C &&
+        if ((this.target == Target.C || this.target == Target.CCPP) &&
```

Comment:
The fact that this has to be repeated everywhere is a bit inelegant, but it is an unavoidable consequence of the CCpp target approach. With the `cmake-force-cpp` proposal this would not have been the case...

Reply:
The conditions became really difficult to parse with the nested or operation. Maybe we could introduce a small method that does the check? For instance, `isCTarget()` or `isCBasedTarget()`